code/modules/mob/living/inventory.dm

```diff
@@ -127,49 +127,37 @@
 	onclose(user, "mob[name]")
 	return
 
-/mob/living/ret_grab(obj/effect/list_container/mobl/L as obj, flag)
-	if ((!( istype(l_hand, /obj/item/weapon/grab) ) && !( istype(r_hand, /obj/item/weapon/grab) )))
-		if (!( L ))
-			return null
-		else
-			return L.container
-	else
-		if (!( L ))
-			L = new /obj/effect/list_container/mobl( null )
-			L.container += src
-			L.master = src
-		if (istype(l_hand, /obj/item/weapon/grab))
-			var/obj/item/weapon/grab/G = l_hand
-			if (!( L.container.Find(G.affecting) ))
-				L.container += G.affecting
-				if (G.affecting)
-					G.affecting.ret_grab(L, 1)
-		if (istype(r_hand, /obj/item/weapon/grab))
-			var/obj/item/weapon/grab/G = r_hand
-			if (!( L.container.Find(G.affecting) ))
-				L.container += G.affecting
-				if (G.affecting)
-					G.affecting.ret_grab(L, 1)
-		if (!( flag ))
-			if (L.master == src)
-				var/list/temp = list(  )
-				temp += L.container
-				//L = null
-				qdel(L)
-				return temp
-			else
-				return L.container
-	return
+/mob/living/ret_grab(var/list/L, var/mobchain_limit = 5)
+	// We're the first!
+	if(!L)
+		L = list()
+	
+	// Lefty grab!
+	if (istype(l_hand, /obj/item/weapon/grab))
+		var/obj/item/weapon/grab/G = l_hand
+		L |= G.affecting
+		if(mobchain_limit-- > 0)
```

Comment:
So this should return up to a linearization of a binary tree structure of depth 5 of people with a unique grab in each hand? No, because the left hand is favored by getting the decrement first.

If the first person has a grab in each hand, the left grab will recurse 4 times, but the right will recurse only 3. In the case where everyone has grabs in each hand, the leftmost path will have depth 4, but the rightmost path will have depth 2. Specifically because you're using decrement and not just `limit - 1`

Reply:
I'm not smart enough with math to understand anything you just said, so you might want to fix that part. It doesn't seem like a big deal to me either way, for a silly 'dance' mechanic.

Reply:
Hopefully a shitty paint drawing (actually gimp, the paint I know how to use would've been simpler but windows 10 removed it because _~quality code!~_) can help explain it. Each box is a unique mob somewhere in the grab chain, the number in the box represents the value of `mobchain_limit` that was passed into the function, and the arrow shows which hand holds the grab, and the mathematical expression being evaluated to produce the value of `mobchain_limit` that's supplied to the recursion.
![image](https://puu.sh/FKDan/d524fcc61f.png)

I should note that this only occurs in cases where a grab is held in both hands, because while it processes the grab in the left hand, it _decrements_ `mobchain_limit`, such that `mobchain_limit` is effectively one less than it is for the left hand, when it gets processed.
This gets even harder to think about in a circular grabchain because recursion, but the mob-uniqueness constraint is useful to establish the underlying issue.

Reply:
Can I just let you fix this? ^^;; I don't think it's really a showstopper, given how bikeshed this sounds.